src/group_theory/index.lean

```diff
@@ -270,4 +271,51 @@ noncomputable def fintype_of_index_ne_zero (hH : H.index ≠ 0) : fintype (G ⧸
 lemma one_lt_index_of_ne_top [fintype (G ⧸ H)] (hH : H ≠ ⊤) : 1 < H.index :=
 nat.one_lt_iff_ne_zero_and_ne_one.mpr ⟨index_ne_zero_of_fintype, mt index_eq_one.mp hH⟩
 
+variables (H K)
+
+/-- Typeclass for finite index subgroups. -/
+class finite_index : Prop :=
+(finite_index : H.index ≠ 0)
+
+/-- Typeclass for finite index subgroups. -/
+class _root_.add_subgroup.finite_index {G : Type*} [add_group G] (H : add_subgroup G) : Prop :=
+(finite_index : H.index ≠ 0)
+
+attribute [to_additive] finite_index
+
+@[priority 100, to_additive] noncomputable instance [hH : finite_index H] : fintype (G ⧸ H) :=
```

Comment:
Now that we have the new `Prop`-valued [finite](https://github.com/leanprover-community/mathlib/blob/1fbe1183eea518b531aca72c62196dbd264a7216/src/data/finite/basic.lean#L83) class I think we should be using it instead of `fintype` wherever we can:
```suggestion
@[priority 100, to_additive] noncomputable instance [hH : finite_index H] : finite (G ⧸ H) :=
```

Reply:
Note that there are several other places in this PR where we could use `finite` instead of `fintype`; I'll leave it to you to locate them.